server/src/server/kb/concept/AttributeImpl.java

```diff
@@ -52,11 +52,13 @@ private AttributeImpl(VertexElement vertexElement, AttributeType<D> type, D valu
     }
 
     public static <D> AttributeImpl<D> create(VertexElement vertexElement, AttributeType<D> type, D value) {
```

Comment:
How can `value` ever have a different `Class` than was is defined in the schema when `type` is defined to be `AttributeType<D>`? This method signature already bounds the type, no? @kasper-piskorski @marco-scoppetta

Reply:
D is `Object` at run-time as generics in java only work at compile-time, there's no way to enforce the specific type at run-time.

Reply:
So if `type` has class `AttributeType<Long>`, the last argument `value` is not bound to `Long` as well?